File: scripts/create_color_images.py
from lib import *
from os.path import dirname, join
from subprocess import run

assets_directory = join(dirname(dirname(__file__)), 'assets')
size = 20
for color in colors:
    image = join(assets_directory, f'{color}.png')
    run([
        'magick',
        '-size',
        f'{size}x{size}',
        f'xc:{colors[color].hex_string}',
        image
    ])
    print(f'Created image {image}.')

File: scripts/create_color_images.rb
require_relative 'lib.rb'

assets_directory = File.join(File.dirname(File.dirname(File.realpath(__FILE__))), 'assets')
size = 20
for key, color in $colors
  image = File.join(assets_directory, "#{key.to_s}.png")
  system "magick -size #{size}x#{size} xc:#{color.hex_string} #{image}"
  puts "Created color image #{image}."
end
